blog-content/articles/protect-your-future-self-by-encapsulating-test-input-preparation.md

```diff
@@ -0,0 +1,127 @@
+---
+date: 2020-01-07
+summary: Examples of how to reduce verbosity and brittleness of your tests by encapsulating input preparation 
+---
+
+# Tests Brittleness and Verbosity slows you down
+
+In [Testing Makes You Faster Day one](entry/testing-makes-you-faster-day-one) article I claimed that writing tests makes you faster.
+However there are test patterns that most likely will slow you down and you need to be ready to handle them. 
+
+Writing tests for a simple logic with a simple input is a breeze. Take for example a test for a simple calculator class. 
+Two numeric inputs and we are done: 
+
+:include-file: com/example/portfolio/SimpleCalculatorTest.java {title: "SimpleCalculatorTest.java"}
+
+However real apps require real domain, and real domain is rarely represented by a couple of numbers.
+Let's take Financial domain as an example. It is full of complex types such as `Transaction` and has a lot of business logic.
```

Comment:
I would rephrase to say 'Let's take Finance as an example'. I feel like 'Financial domain' doesn't flow as nicely.

Reply:
Also, what makes 'Transaction' a complex type? This is not an essential suggestion, but I think it would help if you specified why 'Transaction' is complex.